test/fail/ok/inference.tc.ok

```diff
@@ -37,12 +38,14 @@ inference.mo:111.8-111.17: type error [M0098], cannot apply function of type
   <T <: Int>(x : T) -> T
 to argument of type
   Bool
-because there is no consistent choice for type parameter `T`.
+because there is no way to satisfy subtyping
+  Bool  <:  T  (for argument `x`) 
 inference.mo:112.1-112.10: type error [M0098], cannot apply function of type
   <T <: Int>(x : T) -> T
 to argument of type
   Bool
-because there is no consistent choice for type parameter `T`.
+because there is no way to satisfy subtyping
+  Bool  <:  None  (for argument `x`) 
```

Comment:
Why `None`?
- Expected return type is `()`
- `T` is thus Bivariant, because the expected return type is used as a constraint and no variances are calculated
- Initial `bounds: None <: T <: Int`
- Initial `subs: T <: (@ret : ()), Bool <: (x : T)`
- So first the upper bound `()` is joined with `Int` resulting in `None`
- Current `bounds: None <: T <: None`, which is **not an error** 🤡